src/coreComponents/python/modules/geosx_mesh_doctor/checks/self_intersecting_elements.py

```diff
@@ -0,0 +1,55 @@
+from dataclasses import dataclass
+import logging
+from typing import List
+
+from vtkmodules.vtkFiltersGeneral import (
+    vtkCellValidator
+)
+from vtk.util.numpy_support import (
+    vtk_to_numpy,
+)
+
+from . import vtk_utils
+
+
+@dataclass(frozen=True)
+class Options:
+    tolerance: float
+
+
+@dataclass(frozen=True)
+class Result:
+    jumbled_elements: List[int]
+
+
+def __check(mesh, options: Options) -> Result:
+    valid = 0x0
+    wrong_number_of_points = 0x01
+    intersecting_edges = 0x02
+    intersecting_faces = 0x04
+    non_contiguous_edges = 0x08
+    non_convex = 0x10
+    faces_are_oriented_incorrectly = 0x20
+
+    f = vtkCellValidator()
+    f.SetTolerance(options.tolerance)
+
+    f.SetInputData(mesh)
+    f.Update()
+    output = f.GetOutput()
+
+    cd = output.GetCellData()
+    for i in range(cd.GetNumberOfArrays()):
+        if cd.GetArrayName(i) == "ValidityState":  # TODO change name?
```

Comment:
Does a `SetStateArrayName` (like `SetVolumeArrayName`) function exist?

Reply:
I could not find any... 😢

Reply:
That was my impression too

Reply:
It seems like I'm speaking with a neat IPython user!